storm-core/src/jvm/org/apache/storm/topology/base/BaseWindowedBolt.java

```diff
@@ -66,10 +66,14 @@ public String toString() {
      * Holds a Time duration for time based windows and sliding intervals.
      */
     public static class Duration {
-        public final int value;
+        public final long value;
 
         public Duration(int value, TimeUnit timeUnit) {
-            this.value = (int) timeUnit.toMillis(value);
+            this.value = timeUnit.toMillis(value);
+        }
+
+        public Duration(long value) {
```

Comment:
actually `Duration(long value)` is represent the time interval in milliseconds, so it doesn't time unit to covert .